### PR TITLE
Actions

### PR DESCRIPTION
**Pull request by jburel**
Replace travis by GHA
see https://github.com/jburel/conda-idr-py/actions/runs/427931730

### CODE REVIEW

File: meta.yaml [hunk 1]
```diff
@@ -10,18 +10,18 @@ source:
...
     - VERSION_SUFFIX
 
 requirements:
   host:
     - pip
-    - python =3
+    - python
   run:
-    - python =3
+    - python
```
Comment (manics) [inline meta.yaml:24]: What happens if you don't make this change?

File: meta.yaml [hunk 2]
```diff
@@ -10,18 +10,18 @@ source:
   sha256: e355a7636e5f6883d19955dd3d3edfc4de263040796fad593f4676e49e8a413c
 
 build:
-  noarch: python
-  number: 0
+  # noarch: python
```
Comment (manics) [inline meta.yaml]: Shouldn't this still be `noarch: python`? What are the architecture specific components in this library?

File: meta.yaml (hunk 1 above)
Comment (jburel) [reply meta.yaml:24]: it does install the wrong python version

File: .github/workflows/condabuild.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+          conda build -c ${{ env.channel }} --python ${{ matrix.python }} .
+      - name: List outputs
+        run: |
+          ls -R "${{ steps.condablddir.outputs.condabld }}"
+      - name: Upload noarch
+        uses: actions/upload-artifact@v2
+        with:
+          path: "${{ steps.condablddir.outputs.condabld }}/noarch/idr-py-*"
+          if-no-files-found: error
+        if: startsWith(matrix.os, 'ubuntu')
```
Comment (manics) [inline .github/workflows/condabuild.yml]: I think you need to restrict this to one Python version too.

Comment (jburel) [reply .github/workflows/condabuild.yml]: yep I thought I did, it might have been on another branch

**Review by manics — DISMISSED** [review]

**Comment by manics** [comment]
Have the secrets been setup yet?

**Comment by jburel** [comment]
not yet

File: .github/workflows/condabuild.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        python: [3.6, 3.7, 3.8, 3.9]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      channel:  ome
```
Comment (sbesson) [inline .github/workflows/condabuild.yml:16]:
```suggestion
      channel:  idr
```

File: README.md [hunk 5]
```diff
@@ -1,4 +1,5 @@
 # IDR-py Conda package
 [![Anaconda-Server Badge](https://anaconda.org/idr/idr-py/badges/version.svg)](https://anaconda.org/idr/idr-py)
+[![Actions Status](https://github.com/ome/conda-idr-py/workflows/Conda/badge.svg)](https://github.com/ome/conda-idr-py/actions)
```
Comment (sbesson) [inline README.md]:
```suggestion
[![Actions Status](https://github.com/IDR/conda-idr-py/workflows/Conda/badge.svg)](https://github.com/IDR/conda-idr-py/actions)
```

**Review by sbesson — CHANGES_REQUESTED** [review]

**Review by sbesson — DISMISSED** [review]

**Comment by jburel** [comment]
bumped the version of idr-py.
Secrets added to repo (using omereleases user)

**Review by sbesson — DISMISSED** [review]
Proposing to release with a similar logic as the one used in https://github.com/ome/conda-bftools/pull/6#issuecomment-780450216 (organization token?)

**Comment by jburel** [comment]
I have updated the secrets so the package should end up under the correct user.

**Comment by jburel** [comment]
@sbesson I had to revert 0a1676f44da96f092b86a3c758f59c10d6da2725, This now builds correctly see https://github.com/jburel/conda-idr-py/actions/runs/592533496
If any more comments, I will merge and tag

**Comment by sbesson** [comment]
So is the package going to be pushed to the `ome` or the `IDR` channel?

**Comment by jburel** [comment]
No. This is the confusion
The env.channel is used during the build phase to specify the channel to use to load packages like ``omero-py``
The package will be pushed to ``${{ secrets.ANACONDA_USER }}``

**Comment by sbesson** [comment]
Understood, so `env.channel` determines the channel(s?) required to build the package, in this case `ome` to pull `omero-py`. Am I getting the logic correctly?

**Comment by jburel** [comment]
That's correct

**Review by sbesson — APPROVED** [review]

**Comment by sbesson** [comment]
Now we have a working GitHub Actions conda deployment pipeline, it might be worth looking into merging this with the upstream https://github.com/IDR/idr-py repository. At least for the most simple repositories, I assume it should be feasible to have tags pushing both to PyPI and Anaconda.

**Pull request merged**